docs/interfaces/query-lifecycle.md

```diff
@@ -0,0 +1,181 @@
+
+### Arguments and Route Creation
+
+At this point in the lifecycle, the user has created a CLI command or HTTP Request with all of the data they wish to include in their `Query`. A `CLIContext` exists to assist in the rest of the `Query`'s journey. Now, the next step is to parse the command or request, extract the arguments, create a `queryRoute`, and encode everything.
+
+#### Parse Arguments
+
+In this case, `Query` contains an [address](../core/accounts-fees.md) `delegatorAddress` as its only argument. However, the request can only contain `[]byte`s, as it will be relayed to a consensus engine node that has no inherent knowledge of the application types. Thus, the `CLIContext` `codec` is used to marshal the address as the type [`QueryDelegatorParams`](https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/types/querier.go#L30-L38). All query arguments (e.g. the [`staking`](https://github.com/cosmos/cosmos-sdk/blob/master/docs/spec/staking) module also has [`QueryValidatorParams`](https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/types/querier.go#L45-L53) and [`QueryBondsParams`](https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/types/querier.go#L59-L69)) have their own types that the application `codec` understands how to encode and decode. The module [`querier`](.//building-modules/querier.md) declares these types and the application registers the `codec`s.
+
+Here is what the code looks like for the CLI command:
+
+```go
+delAddr, err := sdk.AccAddressFromBech32(args[0])
+bz, err := cdc.MarshalJSON(types.NewQueryDelegatorParams(delAddr))
+```
+
+Here is what the code looks like for the HTTP Request:
+
+```go
+vars := mux.Vars(r)
+bech32delegator := vars["delegatorAddr"]
+delegatorAddr, err := sdk.AccAddressFromBech32(bech32delegator)
+cliCtx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, cliCtx, r)
+if !ok {
+	return
+}
+params := types.NewQueryDelegatorParams(delegatorAddr)
+```
+
+#### Query Route Creation
+
+Important to note is that there will never be a "query" object created for `Query`; the SDK actually takes a simpler approach. Instead of an object, all the full-node needs to process a query is its `route` which specifies exactly which module to route the query to and the name of this query type. The `route` will be passed to the application baseapp, then module, then [querier](../building-modules/querier.md), and each will understand the `route` and pass it to the appropriate next step. [baseapp](../core/baseapp.md#query-routing) will understand this query to be a `custom` query in the module `staking`, and the `staking` module querier supports the type `QueryDelegatorDelegations`. Thus, the route will be `"custom/staking/delegatorDelegations"`.
```

Comment:
All throughout, it is often unclear if what you describe happens on the user side, or on the full-node size (e.g. query creatino, CLIContext, processing, etc.). This should be made clear at each step (don't hesitate to be explicit)